analyzers/src/SonarAnalyzer.Common/Rules/Hotspots/RequestsWithExcessiveLengthBase.cs

```diff
@@ -59,28 +62,92 @@ protected RequestsWithExcessiveLengthBase(System.Resources.ResourceManager rspec
+        {
+            if (!IsEnabled(context.Options))
+            {
+                return;
+            }
+
+            var attribute = (TAttributeSyntax)context.Node;
+
+            if (attribute.IsKnownType(KnownType.Microsoft_AspNetCore_Mvc_DisableRequestSizeLimitAttribute, context.SemanticModel))
+            {
+                context.ReportDiagnosticWhenActive(Diagnostic.Create(rule, attribute.GetLocation()));
+                return;
+            }
+
+            var requestSizeLimit = IsInvalidRequestSizeLimit(attribute, context.SemanticModel);
+            var requestFormLimits = IsInvalidRequestFormLimits(attribute, context.SemanticModel);
+
+            if ((requestSizeLimit != null
+                 || requestFormLimits != null)
+                && GetMethodOrClassDeclaration(attribute, context.SemanticModel) is { } declaration)
+            {
+                attributesOverTheLimit.AddOrUpdate(
+                    declaration,
+                    new Attributes(requestFormLimits, requestSizeLimit),
+                    (declaration, attributes) => new Attributes(requestFormLimits, requestSizeLimit, attributes));
+            }
+        }
+
+        private void ReportOnCollectedAttributes(CompilationAnalysisContext context, ConcurrentDictionary<SyntaxNode, Attributes> attributesOverTheLimit)
+        {
+            foreach (var attrsByDec in attributesOverTheLimit)
+            {
+                context.ReportDiagnosticWhenActive(
+                    attrsByDec.Value.SecondaryAttribute != null
+                        ? Diagnostic.Create(rule, attrsByDec.Value.MainAttribute.GetLocation(), new List<Location> { attrsByDec.Value.SecondaryAttribute.GetLocation() })
+                        : Diagnostic.Create(rule, attrsByDec.Value.MainAttribute.GetLocation()));
+            }
+        }
 
         private bool IsEnabled(AnalyzerOptions options)
         {
             analyzerConfiguration.Initialize(options);
             return SupportedDiagnostics.Any(d => analyzerConfiguration.IsEnabled(d.Id));
         }
+
+        private struct Attributes
+        {
+            private readonly SyntaxNode invalidRequestForm;
+            private readonly SyntaxNode invalidRequestSize;
+
+            public SyntaxNode MainAttribute =>
+                invalidRequestForm ?? invalidRequestSize;
+
+            public SyntaxNode SecondaryAttribute =>
+                invalidRequestForm != null && invalidRequestSize != null ? invalidRequestSize : null;
+
+            public Attributes(SyntaxNode invalidRequestForm, SyntaxNode invalidRequestSize)
+            {
+                this.invalidRequestForm = invalidRequestForm;
+                this.invalidRequestSize = invalidRequestSize;
+            }
+
+            public Attributes(SyntaxNode invalidRequestForm, SyntaxNode invalidRequestSize, Attributes oldAttributes)
```

Comment:
could you add a test with multiple instances of attributes?

```

    [RequestSizeLimit(8_000_001)]
    [RequestSizeLimit(8_000_002)]
    [RequestSizeLimit(10)]
    [RequestSizeLimit(8_000_003)]
    public class RequestSizeLimitAboveController : Controller
    {

    }
```

Reply:
The build will fail as multiple attributes are not valid, and as this is a Hotspot and it does not run in Sonarlint, Should we cover these cases? The implementation with the struct was based on the assumption not to cover such cases.

Reply:
Ok. Maybe mention it in the code where the `Attribute` struct is used that there is this assumption. Maybe in the future this rule will be migrated and it will help devs to understand the initial assumptions.